gcp_variant_transforms/testing/integration/run_preprocessor_tests.py

```diff
@@ -0,0 +1,178 @@
+               **kwargs):
+    # type: (TestContextManager, str, str, List[str], str, str) -> None
+    super(PreprocessorTestCase, self).__init__(test_name, context.project)
+    self._expected_contents = expected_contents
+    self._report_path = report_path
+    self._keep_reports = context.keep_reports
+    args = ['--input_pattern {}'.format(input_pattern),
+            '--report_path {}'.format(report_path),
+            '--project {}'.format(context.project),
+            '--staging_location {}'.format(context.staging_location),
+            '--temp_location {}'.format(context.temp_location),
+            '--job_name {}'.format(test_name)]
+    for k, v in kwargs.iteritems():
+      args.append('--{} {}'.format(k, v))
+
+    self._pipelines_api_request = {
+        'pipelineArgs': {
+            'projectId': context.project,
+            'logging': {'gcsPath': context.logging_location},
+            'serviceAccount': {'scopes': _SCOPES}
+        },
+        'ephemeralPipeline': {
+            'projectId': context.project,
+            'name': _PIPELINE_NAME,
+            'resources': {'zones': _DEFAULT_ZONES},
+            'docker': {
+                'imageName': context.image,
+                'cmd': ' '.join([_SCRIPT_PATH] + args)
+            }
+        }
+    }
+
+  def validate_result(self):
+    """Validates the results.
+
+    - Checks that the report is generated.
+    - Validates report's contents are the same as ``expected_contents``.
+    """
+    if not filesystems.FileSystems.exists(self._report_path):
+      raise run_tests_common.TestCaseFailure(
+          'Report is not generated in {}'.format(self._report_path))
+    with filesystems.FileSystems.open(self._report_path) as f:
+      for expected_content in self._expected_contents:
+        line = f.readline()
+        if expected_content != line:
+          raise run_tests_common.TestCaseFailure(
+              'Contents mismatch: expected {}, got {}'.format(
+                  expected_content, line))
+    if not self._keep_reports:
+      filesystems.FileSystems.delete(self._report_path)
+
+
+class TestContextManager(object):
+  """Manages all resources for a given run of tests."""
+
+  def __init__(self, args):
+    # type: (argparse.ArgumentParser) -> None
+    self.staging_location = args.staging_location
+    self.temp_location = args.temp_location
+    self.logging_location = args.logging_location
+    self.project = args.project
+    self.credentials = GoogleCredentials.get_application_default()
+    self.image = args.image
+    self.keep_reports = args.keep_reports
+
+  def __enter__(self):
+    return self
+
+  def __exit__(self, *args):
+    return
+
+
+def _get_args():
+  parser = argparse.ArgumentParser()
+  run_tests_common.add_args(parser)
+  parser.add_argument('--keep_reports',
+                      type=bool, default=False, nargs='?', const=True,
+                      help='If set, generated reports and resolved headers are '
```

Comment:
Hmm, I can see that you are deleting the report on line 113 above but where do you delete "resolved headers"?